apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Combobox } from '@headlessui/react';
+import { getObjectFields } from '@mysten/sui.js';
+import clsx from 'clsx';
+import { useState } from 'react';
+
+import { ReactComponent as SearchIcon } from '~/assets/SVGIcons/24px/Search.svg';
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import { useGetObject } from '~/hooks/useGetObject';
+import { DisclosureBox } from '~/ui/DisclosureBox';
+import { AddressLink, ObjectLink, TransactionLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '~/ui/Tabs';
+import { Text } from '~/ui/Text';
+import { ListItem, VerticalList } from '~/ui/VerticalList';
+
+interface ObjectFieldsProps {
+    id: string;
+}
+
+interface FieldItemProps<T> {
+    value: T;
+    type?: string;
+}
+
+function FieldItem<T>({ value, type }: FieldItemProps<T>) {
```

Comment:
Would _probably_ break this out into it's own file just to keep things a little cleaner